elastic/eui
# [EuiCollapsibleNavBeta] Build out component state and responsive behavior

## Summary

- ⚠️ This PR is part **1 of 2** of completing EuiCollapsibleNavBeta's responsive/docked behavior.
  - The part that is missing is the part where `EuiCollapsibleNavItem` needs to conditionally render a icon that toggles a popover, but only for larger desktop screens when in the "collapsed" state.
  - I opted to split out this work into separate PRs, as this one was already getting pretty long in the tooth in terms of LOC/review time.
- Parent issue: https://github.com/elastic/eui/issues/6759
- Previous PRs: https://github.com/elastic/eui/pull/6904

As always, I recommend [following along by commit](https://github.com/elastic/eui/pull/7027/commits).

### Screenshots

#### Desktop - expanded
<img width="400" alt="" src="https://github.com/elastic/eui/assets/549407/6cd6b097-3d34-4adf-9817-f46e35289414">

#### Desktop - collapsed
As noted above, this is **expected to be broken** and will be implemented in the next PR. However, the flyout width should be correct.

<img width="200" alt="" src="https://github.com/elastic/eui/assets/549407/7f1b0406-f1aa-4485-bcad-a093b3ec27e0">

<img width="100" alt="" src="https://github.com/elastic/eui/assets/549407/21b7847f-06ea-4238-b485-3c9c57cf9d9f">

#### Mobile - expanded
<img width="400" alt="" src="https://github.com/elastic/eui/assets/549407/d0d80ddc-9a45-4334-be9a-6496c50d8eee">

Smallest screens should have the flyout overlay to 100% width
<img width="300" alt="" src="https://github.com/elastic/eui/assets/549407/68ea2ebe-ff7c-4c55-9e74-4330013c8fd1">

#### Mobile - collapsed
<img width="400" alt="" src="https://github.com/elastic/eui/assets/549407/2868d3fd-e53a-4497-ae91-29b7af5a2ba0">

## QA

- `gh pr checkout 7027`
- `yarn storybook`
- Go to http://localhost:6006/?path=/story/euicollapsiblenavbeta--kibana-example
- View the demo on desktop
    - [x] Confirm that clicking the collapse button sets the flyout width correctly (should be same as the width of the toggle button)
    - [x] Confirm the toggle button toggles between left/right menu icons
    - [x] Confirm that changing the `side` prop in the Storybook controls panel correctly toggles the flyout position and the icon directions
- View the demo on large mobile
    - [x] Confirm the flyout is now an overlay 
    - [x] Confirm the icons now toggle between a menu and a cross
- View the demo on small mobile
    - [x] Confirm the flyout is now full-width over the entire screen
- In the Storybook controls panel, set the `width` prop to a custom width, e.g. `400`
- [x] Resize your window appropriately to confirm that the nav's breakpoints are dynamic based on the width passed - it's considered "desktop" at 3x the width of the nav, and "smallest" at under 1.5x the width of the nav
- Go to http://localhost:6006/?path=/story/euicollapsiblenavbeta--multiple-fixed-headers
- [ ] Confirm the header has correctly accounted and positioned for both fixed headers

### General checklist

~- [ ] Checked for **accessibility** including keyboard-only and screenreader modes~
  - NOTE: This isn't quite yet ready due to the component being merged in in-progress states. I plan on making a final a11y/screenreader pass at the end / after the final PR

Below items are all N/A due to the component's beta status
~- [ ] A **[changelog](https://github.com/elastic/eui/blob/main/wiki/contributing-to-eui/documenting/changelogs.md)** entry exists and is marked appropriately~
~- [ ] Checked for **breaking changes** and labeled appropriately~
~- [ ] Props have proper **autodocs** (using `@default` if default values are missing) and **[playground toggles](https://github.com/elastic/eui/blob/main/wiki/contributing-to-eui/documenting/playgrounds.md)**~
~- [ ] Added **[documentation](https://github.com/elastic/eui/blob/main/wiki/contributing-to-eui/documenting)**~
~- [ ] Checked **[Code Sandbox](https://codesandbox.io/)** works for any docs examples~
~- [ ] Updated the **[Figma](https://www.figma.com/community/file/964536385682658129)** library counterpart~
